controllers/azuremachine_reconciler.go

```diff
@@ -334,3 +335,21 @@ func (s *azureMachineService) isAvailabilityZoneSupported() bool {
 	s.machineScope.V(2).Info("Availability Zones are not supported in the selected location", "location", s.machineScope.Location())
 	return azSupported
 }
+
+// Pick image from the machine configuration, or use a default one.
+func getVMImage(scope *scope.MachineScope) infrav1.Image {
+	imageSpec := scope.AzureMachine.Spec.Image
+	// Use custom Marketplace image, Image ID or a Shared Image Gallery image if provided
+	if imageSpec.ID != nil || imageSpec.Gallery != nil || (imageSpec.Publisher != nil && imageSpec.Offer != nil && imageSpec.SKU != nil && imageSpec.Version != nil) {
+		return imageSpec
+	}
+	// For now this assumes Ubuntu 18.04. It can be extended once we support other OSes.
+	k8sVersion := to.String(scope.Machine.Spec.Version)
+	sku := azure.GetDefaultImageSKUID(k8sVersion)
```

Comment:
Better to create a function GetDefaultImage(k8sVersion) so defaults are encapsulated in single place

so return would look like
```
return GetDefaultImage(scope.Machine.Spec.Version)
```